sdk-cpp/tests/download_tests.cpp

```diff
@@ -532,19 +532,22 @@ TEST_F(DownloadTests, SimpleBlockingDownloadTest_ClientNotRunningPortFilePresent
     ASSERT_FALSE(boost::filesystem::exists(g_tmpFileName));
 }
 
-
 TEST_F(DownloadTests, MultipleRestPortFileExists_Download)
 {
-// Enable after we have the ability to start the daemon after creating rest port files.
-// This will ensure that the actual rest port file will have the latest timestamp.
-#if 0
+    TestHelpers::StopService("deliveryoptimization-agent.service");
+    auto startService = dotest::util::scope_exit([]()
+        {
+            TestHelpers::StartService("deliveryoptimization-agent.service");
+        });
     TestHelpers::CreateRestPortFiles(5);
+    ASSERT_GE(TestHelpers::CountRestPortFiles(), 5u);
+    std::this_thread::sleep_for(std::chrono::seconds(2));
+    startService.reset();
+    std::this_thread::sleep_for(std::chrono::seconds(2));
+    ASSERT_EQ(TestHelpers::CountRestPortFiles(), 1u) << "All other restport files must be deleted by the agent";
```

Comment:
Could also check to see if the portfile created has a newer timestamp than the prexisting ones, to ensure that the file created was by the daemon upon restart.

Reply:
If it wasn't created by the daemon, the download will fail right?

Reply:
true